### ISSUE TITLE
Real Estate Changes

### ISSUE THREAD

**Issue by ImArteria** [InZidiuZ/op-framework-issue-tracker#2172]
**Assurance:**
Have you read through the rules from the `README.md` file in the root folder of this GitHub repository?
Yes

**Summary:**
So in short, me personally from hearing around, Real Estate is somewhat of a little bit of hard roleplay for people to get into. There's a change that I think needs to be made that would encourage new roleplayers to get into the real estate line of work, while opening up other new roleplay opportunities but keeping the real estate business solid.

I'm sure it's been suggested before, but Real Estate should be solo-house based instead of block based. I know the way real estate works now, a realtor would have to purchase one block for ___ amount of money. Each week, assuming no one rents a house, the realtor has to pay for maintenance fees until the house is rented. Some blocks have 10+ mansions on which can cost to upwards of 100+k..

My overall suggestion, which whatever is easier developing wise, would be making real estate a "per house" type of thing instead of "per block". 

So this could work in one of two ways, whichever is easier for development. 
- A realtor could buy a "house" (like a block) and pay maintenance fee, which may be easier for coding since realtors already have houses purchased. Once the home is forfeited, the owner pays the maintenance fee and has that house to forever sell to another customer.
- A realtor no longer pays a maintenance fee, but instead can go up to any house and sell it to an individual. With this being said, the house costs would have to be split 50/50. 50 of it goes to State, 50 of it goes to the realtor. Of course this could work however it pleased, it doesn't have to be 50/50, it would just an example number.

**Reason:**
Why would your suggestion be a good change to the framework?

It would bring RP all around. It would encourage new players to become realtors, it would give more legal people (like lawyers, PD, EMS, etc) opportunities to be realtors as well because State would open up more realtor spots due to it not being inflated and give people more chances. It would allow people to actually choose what houses they want, instead of "Oh I have to buy a house that this realtor has a block for". It would also give new players opportunities to live somewhere that isn't owned by an entire gang or whatnot.

**Comment by vol5** [comment]
> I appreciate the reasoning but disagree with the suggestion. 

I carry the same sentiment as MetaHatter after reading this. To me the current system is solid but only needs improvements to what we have not a complete overhaul. I dislike the more realistic approach to house selling. To many variables come into play with the way it would be.

**Comment by coalaura** [comment]
I think this idea is way better than the current system. Right now realtors sometimes struggle to buy whole blocks due to the price. It would be a lot easier for groups to live next to each other if they would just have to purchase the houses they actually need and not try to fill a whole block just to convince a realtor to buy it

**Comment by xplrerin** [comment]
It would be nice to buy individual houses if the /comp was a lot cleaner. If we could for example, Say to state I want to buy Paleto Boulevard 1 and pay the maintenance so $3k for a Lesters and just have individual houses on the list for us to sell, sure is cheaper.. But would be very messy on the /comp. 

Some blocks just need to be reduced in size, for example North Rockford Drive or Del Perro Freeway. Current Realtors have had a meeting in the past and agreed if these blocks were even just halved in size, it'd be a lot easier. Another thing, if realtors could choose which interior a house had, so we can discuss with our customers which they'd prefer and choose.. That would reduce the amount of blocks having to be purchased.

**Comment by vol5** [comment]
I still strongly feel an overhaul as described in the suggestion should not be fulfilled. 

This problem here 
> Right now realtors sometimes struggle to buy whole blocks due to the price. It would be a lot easier for groups to live next to each other if they would just have to purchase the houses they actually need and not try to fill a whole block just to convince a realtor to buy it

Can be solved with as @xplrerin mentioned above.
> Some blocks just need to be reduced in size,

@MetaHatter and I talked about an extensive block audit a while back which has not been done. This would allow for downsizing of a lot of the problematic spots. T would you be willing to entertain that as a solution for the problem you mentioned? 

I just don't feel that there is anything inherently wrong with the core of the current system, especially when purchasing should not be an issue considering the economy right now and how house prices have been static.

**Comment by coalaura** [comment]
Choosing what interior a house has makes 0 sense.

**Comment by xplrerin** [comment]
I don't see why it makes no sense - Would be a lot easier for someone to say "Hey I like this house on this street" and there be a 2/3 interiors viable for the interior. I've seen full ass houses have trailer interiors, I'd say that made less sense. Or the low-end apartments when the house is on the beach (The case with most of the Paleto housing).. I'm not saying a trailer has to have a Michael's interior, just with some of the properties.

**Comment by coalaura** [comment]
> I don't see why it makes no sense - Would be a lot easier for someone to say "Hey I like this house on this street" and there be a 2/3 interiors viable for the interior. I've seen full ass houses have trailer interiors, I'd say that made less sense. Or the low-end apartments when the house is on the beach (The case with most of the Paleto housing).. I'm not saying a trailer has to have a Michael's interior, just with some of the properties.

No, the interiors are chosen based on the size of the house. Those wont ever be selectable by the realtor.

**Comment by Masik52** [comment]
> Some blocks just need to be reduced in size, for example North Rockford Drive or Del Perro Freeway. Current Realtors have had a meeting in the past and agreed if these blocks were even just halved in size, it'd be a lot easier. Another thing, if realtors could choose which interior a house had, so we can discuss with our customers which they'd prefer and choose.. That would reduce the amount of blocks having to be purchased.

I do think the current block system is fine the way it is.  Sure being able to sell an individual house would be great and less of a headache for the realtor but I agree it would take most of the risk out of being a company owner.  I completely agree that some blocks need to be adjusted.  I personally have 2 of the larger blocks under my company which I've managed to keep full for quite some time now.  From the international side of things I think there's several blocks that have likely never been owned simply due to the sheer volume of houses on the block.  Those would be Del Perro Freeway, Imagination Court, Magellan Avenue and Invention Court.  These blocks have 44, 40, 29 and 33 units in the block respectively.  

Del Perro Freeway is literally about 1/4 the size of my entire housing portfolio in one block alone.  I know a bunch of these places have multiple units in 1 building so there's a couple options available that could be done to make the blocks a bit more attractive to own by companies.  Del Perro Freeway has a natural divide at Del Perro Freeway 13 / Apt1 or Apt2 and Del Perro Freeway 14 that could be the division point if you were to split that block up into 2 blocks rather than 1.  That would give Del Perro Freeway A 18 units leaving Del Perro Freeway B with 26 and those would be a lot more feasible to manage while still having risk involved.  Another option would be to eliminate the multiple units per building on these.  Using Del Perro Freeway again as the guinea pig it currently has 44 units on the block if you removed the apt1/apt2/apt3's and just made each building a single unit you've reduced the block size down to 27 rather than 44.  If you did both of these suggestions on removing the multi units and splitting the block at the natural divide you'd have a block consisting of 13 and 14 houses and I can almost assure you they'd be obtained by a realtor immediately.  
![Del Perro Fwy](https://user-images.githubusercontent.com/52395441/218537787-3363c43b-000d-40d9-835b-46bc49b78a60.png)


Magellan Avenue has natural break points as well if you were to split up that block into multiple blocks.  Magellan Avenue 1 - 10 would be a block of 10 units. Magellan Avenue 11-18 would be a block of 8 units and then Magellan Avenue 19 a small 5 unit block and Magellan Avenue 20-21 would also be a small 5 unit block where each would have break points for the block that make sense and would be extremely manageable by any real estate company.
![Magellan Court](https://user-images.githubusercontent.com/52395441/218537247-e9c867ba-3cc4-43eb-b338-0167dcfcada8.png)


Imagination Court has the natural break points again where you could break that one big block up into 3 different blocks.  Imagination Court 1-9 would be a 12 unit block, Imagination Court 10-19 would be a 14 unit block, Imagination Court 20-29 would be a 14 unit block that has clumps of houses that would make sense as to why they'd be in a block together.  These would also be very manageable by any real estate company.
![Imagination Court](https://user-images.githubusercontent.com/52395441/218535589-7fb4166c-ae78-4cdb-8b1c-cd8dd542a05f.png)

Last but not least Invention Court following the same reasoning as above this block could be broken up into 4 blocks that would make sense with how the houses are clumped together.  Invention Court 1-8 would be a 11 unit block, Invention Court 9-17 would be a 9 unit block, Invention Court 18-26 would be a 10 unit block and finally Invention Court 27-28 would be a 3 unit block.  Again all of these would be extremely manageable by a real estate company to allow for lower to mid end price pointed houses to be available in areas that could be desirable for multiple reasons.
![Invention Court](https://user-images.githubusercontent.com/52395441/218536283-d5ef8057-02e2-436c-ae34-4dd5fee3b0ba.png)

That being said I'm not sure if these blocks are owned by any companies in any of the subdivisions so that could throw a wrench into the reduction in size of the blocks if that route was leaned towards.  Even if they are owned by companies in the subdivisions if the blocks were broken up there would be no impact to the current player base the realtors would just have more blocks under their company.

**Comment by vol5** [comment]
>> I do think the current block system is fine the way it is. Sure being able to sell an individual house would be great and less of a headache for the realtor but I agree it would take most of the risk out of being a company owner. I completely agree that some blocks need to be adjusted. I personally have 2 of the larger blocks under my company which I've managed to keep full for quite some time now. 

The way it has just been outlined by @Masik52 hits the nail on the head for changes that would be most beneficial and impactful to the current system. As a realtor on international I do agree that the core of the current block system is fine the way it is and individual would remove risk from being a company owner as Masik and Meta mentioned above.

**Comment by coalaura** [comment]
We're not gonna split up any blocks. Doing so would make everyone in half that block loose their houses (this includes all subdivisions). If we ever end up rewriting that script it will most likely be changed to purchasing individual houses and get rid of blocks entirely

**Comment by navonody1** [comment]
If splitting up the blocks isn't an option, is it possible to make it so that realtors aren't charged for houses that are empty on blocks they have? Or would that also involve rewriting the script? Realtors are an extension of State, assisting them in housing people of Los Santos FOR FREE essentially. Any profits realtors make should just be considered payment for their efforts IMO. That being said, I don't think realtors should be penalized for not being able to fill up a block completely, especially with the way the current system is set up.

**Comment by ImArteria** [comment]
If realtors aren't charged for houses that are empty than that would eliminate ALL risk which is the whole argument of splitting them.

**Comment by navonody1** [comment]
> If realtors aren't charged for houses that are empty than that would eliminate ALL risk which is the whole argument of splitting them.

Right, but it was suggested that we're not splitting up the blocks. T stated that if the script is rewritten, it would involve purchasing individual homes which also wouldn't involve risk for any realtor...

Also, do realtors in real life ever risk losing money? If risk is mitigated as you say, I don't see the issue with that.

**Comment by xplrerin** [comment]
As far as I know especially on the international server, government to fund us for empty houses on bigger blocks if we’ve bought a bigger block for an absurd reason. For the most part, the bigger blocks minus grove street and north Rockford drive aren’t purchased- and I’d hope if anyone wanted to purchase them they’d think carefully beforehand, at least till a new script is written.

Whilst we’re on the real estate improvement topic again, I’d love there to be a way when civs do /blocks it only shows what realtor that has the block and remove available spaces but still keep the total spaces. this way they dont know if its full or not. After speaking to a few realtors in our business discord, it seems that people do check /blocks and it takes away from the roleplay, as the blocks command will tell them a available spot and pricing - on gang blocks etc , it is incredibly frustrating. People should reach out to find out availability instead of abusing the command - even better if the command could only be used by a licensed realtor. (Thus allowing people to have to reach out to a realtor to find out info on a block)

**Comment by ayushpatel1007** [comment]
If we allow buying single houses then it can be a disaster also. This is my personal opinion so i can be wrong.

So Allowinn to buy single house will be great for 1st company opened in server. But there are many times when many comes open like months of months later then other companies and that last 1-2 comapny will have to buy the worst houses that no one boought.

Currently with block system, if the owner is using his mind then he sell the entire block with great stratetic decisions , ideas , promotions etc. 

IF we allow buying single house then no one will ever buy michale mansion. Also no one will prefer houses/mansion with no big pools or gardens. It will only good for the first company which is opened in server who will bu only the best mansions of city and leave other mansion that are not good that no one would love to buy.

example: mansions of north rockford drive block are good but are they good as richman street mansions with biggest area a mansion can have in server ? then why would even a company will think of  buying mansion of blocks like wild oats drive , kimble hill or any other mansions?

specially michale mansion are least loved in mansion category, everyones prio is to buy high end apartment or franklin and then nothing is left then they buy michale.

solution to this can be block system will be same but we dont have to pay maintance of house that are empty, that way companies will able to afford the house and can sell without having stress of selling empty houses. also many alternative can be thought off.

**Comment by M3Mproperty1** [comment]
Real estate cycles in India can be influenced by various factors like economic conditions and market trends. For optimal results, consider purchasing a home during a buyer's market when prices are relatively lower. Keep an eye on[ M3M Properties](https://www.m3mantalya-hills.com/), a reputed real estate developer, for potential investment opportunities. Timing your purchase during a market downturn or when interest rates are favorable can enhance your chances of getting a good deal. Conduct thorough research and consult with experts to make an informed decision aligning with both market conditions and M3M Properties offerings.

**Comment by coalaura** [comment]
https://github.com/InZidiuZ/op-framework-issue-tracker/issues/3215

**Comment by ayushpatel1007** [comment]
Block offer for Gang/family

Sending offer for whole block to family/gang. Genrally gangs buy whole block as their gang hood and we have rule in legacy india that one person can be owner of only 2 house for single block. That’s means you can buy only 2 house in same block and if you want more than buy to another block. (Mostly this rule is for gang hoods and maybe doesn’t apply on residents who are civilian )


/block offer , can new command that allow to send offer to gang owner for block in which obviously their can be different house owners but the rent of entire block will be paid together . So for example if MG bought gang hood Fudge lane then they have to buy that block from company by /block offer and then they can select owners of house but rent will be paid by only one person of entire block so if rent of entire block is 100k then they have to pay all together. 


why we need this?
We have noticed that whenever any family buys block. After few weeks , they forget to pay rent or didn’t pay rent of few house and then because rent is over due , their house access is revoked (evicted) now as they paid only for half of house, they don’t take seriously sometimes that there half block is empty and half they are holding. Ultimately this create loss for company. 


as it is gang hood, company can sell that house to civilian because gang will just keep killing civilians in their block so company don’t have option to give empty houses to civilians but gang members

**Comment by InZidiuZ** [comment]
This issue tracker is being shut down and replaced by Discord threads. If this issue is still relevant, create a thread on [Discord](https://discord.gg/opfw).
